assembly.py

```diff
@@ -333,7 +333,7 @@ def parser_assemble_trinity(parser=argparse.ArgumentParser()):
         default=tools.trinity.TrinityTool.jvm_mem_default,
         help='JVM virtual memory size (default: %(default)s)'
     )
-    parser.add_argument('--threads', default=1, help='Number of threads (default: %(default)s)')
+    parser.add_argument('--threads', type=int, default=None, help='Number of threads (default: all available cores)')
```

Comment:
Why don't we set the default cores == ALL_CORES at the parser level?

Reply:
I think I like the idea of using `None`, partly because some of these tools already do some form of autodetect on their own when a parameter isn't specified... but some don't.. so I could go either way.